### PR TITLE
Add different types of RzGraphNodeInfos.

### PR DESCRIPTION
**Pull request by Rot127**
**Your checklist for this pull request**
- [x] I've read the [guidelines for contributing](https://github.com/rizinorg/rizin/blob/master/DEVELOPERS.md) to this repository
- [x] I made sure to follow the project's [coding style](https://github.com/rizinorg/rizin/blob/master/DEVELOPERS.md#code-style)
- [x] I've documented or updated the documentation of every function and struct this PR changes. If not so I've explained why.
- [x] I've added tests that prove my fix is effective or that my feature works (if possible)
- [ ] I've updated the [rizin book](https://github.com/rizinorg/book) with the relevant information (if needed)

**Detailed description**

This adds support for multiple graph node infos.
It removes the assumption that information of
a graph node must be some strings and an offset.

**Test plan**

All green.
And more to ccome for CFG and iCFGs

**Closing issues**

None

### CODE REVIEW

**Review by ret2libc — CHANGES_REQUESTED** [review]
Can we add at least a unit test for these new graph features?

**Comment by Rot127** [comment]
Yes, of course. It's on the list above. Going to add them tomorrow.

File: librz/core/cgraph.c [hunk 1]
```diff
@@ -911,16 +978,22 @@ static inline bool ignore_next_instr(const RzAnalysisOp *op) {
  * \return true On success.
  * \return false On failure.
  */
-static bool add_edge_to_cfg(RZ_NONNULL RzGraph /*<RzGraphNodeInfo *>*/ *graph, RZ_NONNULL RzVector /*<ut64>*/ *to_visit, RZ_NONNULL HtUU *nodes_visited, ut64 from, ut64 to) {
-	rz_return_val_if_fail(graph && to_visit && nodes_visited, -1);
+static bool add_edge_to_cfg(RZ_NONNULL RzGraph /*<RzGraphNodeInfo *>*/ *graph,
```
Comment (XVilka) [inline librz/core/cgraph.c:981]: Please update Doxygen too since you renamed parameters

File: librz/util/graph_drawable.c [hunk 2]
```diff
@@ -5,29 +5,112 @@
...
+		return NULL;
+	case RZ_GRAPH_NODE_TYPE_DEFAULT:
+	case RZ_GRAPH_NODE_TYPE_CFG:
+	case RZ_GRAPH_NODE_TYPE_ICFG:
+		break;
+	}
+	return info;
+}
+
 RZ_API void rz_graph_free_node_info(void *ptr) {
```
Comment (XVilka) [inline librz/util/graph_drawable.c]: Add NULLABLE here pls

File: librz/util/graph_drawable.c [hunk 3]
```diff
@@ -5,29 +5,112 @@
...
+ *
+ * \param title The title describing the node.
+ * \param body The body text describing the node.
+ * \param offset A numeric offset of this node. 0 if invalid.
+ *
+ * \return The initialized RzGraphNodeInfo or NULL in case of failure.
+ */
+RZ_API RzGraphNodeInfo *rz_graph_create_node_info_default(const char *title, const char *body, ut64 offset) {
+	RzGraphNodeInfo *data = RZ_NEW0(RzGraphNodeInfo);
+	if (data) {
```
Comment (XVilka) [inline librz/util/graph_drawable.c]: Invert this condition to avoid unnecessary nesting

File: librz/util/graph_drawable.c [hunk 4]
```diff
@@ -5,29 +5,112 @@
...
+ *
+ * \param address The address of the instruction this node represents.
+ * \param call_target_addr The address of the procedure called, if this node is a call.
+ * \param flags Additional flags which describe the node.
+ *
+ * \return The initialized RzGraphNodeInfo or NULL in case of failure.
+ */
+RZ_API RzGraphNodeInfo *rz_graph_create_node_info_cfg(ut64 address, ut64 call_target_addr, RzGraphNodeType type, RzGraphNodeSubType subtype) {
 	RzGraphNodeInfo *data = RZ_NEW0(RzGraphNodeInfo);
 	if (data) {
```
Comment (XVilka) [inline librz/util/graph_drawable.c]: Invert this condition to avoid unnecessary nesting

File: librz/util/graph_drawable.c [hunk 5]
```diff
@@ -5,29 +5,112 @@
...
+ * \brief Initializes a node info struct of an iCFG node.
+ *
+ * \param address The address of the procedure this node represents.
+ * \param flags Additional flags which describe the node.
+ *
+ * \return The initialized RzGraphNodeInfo or NULL in case of failure.
+ */
+RZ_API RzGraphNodeInfo *rz_graph_create_node_info_icfg(ut64 address, RzGraphNodeType type, RzGraphNodeSubType subtype) {
+	RzGraphNodeInfo *data = RZ_NEW0(RzGraphNodeInfo);
+	if (data) {
```
Comment (XVilka) [inline librz/util/graph_drawable.c]: Invert this condition to avoid unnecessary nesting

**Review by XVilka — CHANGES_REQUESTED** [review]
Few minor nitpicks but overall LGTM

**Review by XVilka — APPROVED** [review]

**Comment by XVilka** [comment]
@Rot127 one minor thing left:
```
Mismatched function annotation for rz_graph_free_node_info at <rizin/librz/include/rz_util/rz_graph_drawable.h:82:13> : None (was RZ_NULLABLE at <rizin/librz/util/graph_drawable.c:31:13>
```

**Pull request merged**